packages/instance-model/lib/model.js

```diff
@@ -135,13 +139,75 @@ const InstanceModel = AmpersandModel.extend(
           return ['fetching', 'refreshing'].includes(this.refreshingStatus);
         },
       },
+      isTopologyWritable: {
+        deps: ['topologyDescription.type'],
+        fn() {
+          return TopologyType.isWritable(this.topologyDescription.type)
+        }
+      },
+      singleServerType: {
+        deps: ['topologyDescription.type', 'topologDescription.servers'],
+        fn() {
+          if (this.topologyDescription.type === TopologyType.SINGLE) {
+            return this.topologyDescription.servers[0].type;
+          }
+          return null;
+        }
+      },
+      isServerWritable: {
+        deps: ['singleServerType'],
+        fn() {
+          return this.singleServerType !== null && ServerType.isWritable(this.singleServerType);
+        }
+      },
+      isWritable: {
+        deps: ['topologyDescription.type', 'isTopologyWritable', 'isServerWritable'],
+        fn() {
+          if (this.isTopologyWritable) {
+            if (this.topologyDescription.type === TopologyType.SINGLE) {
+              return this.isServerWritable;
+            } else {
+              return true;
+            }
+          } else {
+            return false;
+          }
+        }
+      },
+      description: {
+        deps: ['topologyDescription.type', 'isTopologyWritable', 'isServerWritable', 'singleServerType'],
+        fn() {
+          const topologyType = this.topologyDescription.type;
+
+          if (this.isTopologyWritable) {
+            if (topologyType === TopologyType.SINGLE) {
+              const message = this.isServerWritable ? 'is writable' : 'is not writable';
+              return `Single connection to server type: ${ServerType.humanize(this.singleServerType)} ${message}`;
+            }
+            return `Topology type: ${TopologyType.humanize(topologyType)} is writable`;
+          }
+          return `Topology type: ${TopologyType.humanize(topologyType)} is not writable`;
+        }
+      },
+      env: {
```

Comment:
We have env and then various isAtlas, isDataLake, etc things all over the codebase, unfortunately. But at least for now we can know this as soon as we get the instance details. We should probably pass it to the collection tab plugins when creating their stores.